src/workerd/jsg/jsg.h

```diff
@@ -1770,14 +1772,10 @@ class ContextGlobal {
 
   KJ_DISALLOW_COPY_AND_MOVE(ContextGlobal);
 
-  ModuleRegistry& getModuleRegistry() {
-    return *moduleRegistry;
-  }
-
  private:
-  kj::Own<ModuleRegistry> moduleRegistry;
+  kj::Own<void> moduleRegistry;
 
-  void setModuleRegistry(kj::Own<ModuleRegistry> registry) {
+  void setModuleRegistry(kj::Own<void> registry) {
     moduleRegistry = kj::mv(registry);
   }
```

Comment:
Reviewer note: Some context on this change... this field in `ContextGlobal` was already storing the original module registry instance, changing this to a `kj::Own<void>` allows us to use it to store either the new one or the old. Since a context is always going to be one or the other, we only need to store it once.

We do not need the `getModuleRegistry` method on the `ContextGlobal`, however, since we can acquire the pointer to whatever module registry is used via the embedder slots on the `v8::Context`. Essentially this unifies how we store and keep the module registry instance alive.

When using the new module registry, what this slot is actually storing is the `kj::Own<IsolateModuleRegistry>`.

Reply:
the opaque pointers here aren't how we access it, the `kj::Own<void>` is only how we keep it alive. The actual interfaces are accessed via the aligned pointers in the `v8::Context` and while the actual registry interfaces themselves are different there are apis that abstract their use (see `resolveModule` on `jsg::Lock`).

Reply:
Yep I understand, I'd still love to see it have a type, but otherwise we can give this a better name.
How about `kj::Own<const void> registryBackingOwner`.
We use this pattern elsewhere as well.

Reply:
renamed and added a commment